drivers/spi/spi_nrfx_spis.c

```diff
@@ -105,42 +103,36 @@ static int configure(const struct device *dev,
 	return 0;
 }
 
-static void prepare_for_transfer(const struct device *dev)
+static void prepare_for_transfer(const struct device *dev,
+				 const uint8_t *tx_buf, size_t tx_buf_len,
+				 uint8_t *rx_buf, size_t rx_buf_len)
 {
 	struct spi_nrfx_data *dev_data = get_dev_data(dev);
 	const struct spi_nrfx_config *dev_config = get_dev_config(dev);
-	struct spi_context *ctx = &dev_data->ctx;
 	int status;
 
-	size_t buf_len = spi_context_max_continuous_chunk(ctx);
-
-	if (buf_len > 0) {
+	if (tx_buf_len > dev_config->max_buf_len ||
+	    rx_buf_len > dev_config->max_buf_len) {
+		LOG_ERR("Invalid buffer sizes: Tx %d/Rx %d",
+			tx_buf_len, rx_buf_len);
+		status = -EINVAL;
+	} else {
 		nrfx_err_t result;
 
-		if (buf_len > dev_config->max_buf_len) {
-			buf_len = dev_config->max_buf_len;
-		}
-
-		result = nrfx_spis_buffers_set(
-			&dev_config->spis,
-			ctx->tx_buf,
-			spi_context_tx_buf_on(ctx) ? buf_len : 0,
-			ctx->rx_buf,
-			spi_context_rx_buf_on(ctx) ? buf_len : 0);
+		result = nrfx_spis_buffers_set(&dev_config->spis,
+					       tx_buf, tx_buf_len,
+					       rx_buf, rx_buf_len);
```

Comment:
Same here, and the assignment to result is not needed as it can be replaced with direct comparison to `NRFX_SUCCESS`.